src/test/java/com/epam/dataservice/AccidentBatchLoaderTest.java

```diff
@@ -0,0 +1,197 @@
+
+        System.out.println("Starting 2 readers");
+        executor.execute(readerTask1);
+        executor.execute(readerTask2);
+
+        readerTask1.get();
+        readerTask2.get();
+
+        endDatLoading(executor);
+    }
+
+    private static void endDatLoading(ExecutorService executor){
+        executor.shutdown();
+        try {
+            executor.awaitTermination(5, TimeUnit.MINUTES);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        System.out.println("Data loading finished");
+    }
+
+    public static void handleMultipleFilesWithCallable() throws ExecutionException, InterruptedException {
+
+        LinkedList<String> fileQueue = new LinkedList<>();
+        fileQueue.add(filePath3);
+        fileQueue.add(filePath4);
+        fileQueue.add(filePath1);
+        fileQueue.add(filePath2);
+
+        ExecutorService executor = Executors.newFixedThreadPool(4);
+
+        /**
+         * We want to process 2 files at a time so using 2 blocking queue
+         * to hold the result
+         */
+        BlockingQueue<List<RoadAccident>> dataQueue1 = new ArrayBlockingQueue<List<RoadAccident>>(2);
+        BlockingQueue<List<RoadAccident>> dataQueue2 = new ArrayBlockingQueue<List<RoadAccident>>(2);
+
+
+        //start consumer threads
+        executor.execute(new DataConsumerNonEnding(dataQueue1, "dataQueue1"));
+        executor.execute(new DataConsumerNonEnding(dataQueue2, "dataQueue2"));
+        System.out.println("Started Consumers");
+
+        //Create 2 reader task and start
+        FutureTask<Integer> readerTask1 = new FutureTask<Integer>(new AccidentBatchLoader(40000,dataQueue1,fileQueue.remove()));
+        FutureTask<Integer> readerTask2 = new FutureTask<Integer>(new AccidentBatchLoader(40000,dataQueue2,fileQueue.remove()));
+
+        System.out.println("Starting 2 readers");
+        executor.execute(readerTask1);
+        executor.execute(readerTask2);
+
+        while(!fileQueue.isEmpty()){
+            if(readerTask1.isDone()){
+                readerTask1 = new FutureTask<Integer>(new AccidentBatchLoader(40000,dataQueue1,fileQueue.remove()));
+                executor.execute(readerTask1);
+            }
+            if(readerTask2.isDone()){
+                readerTask2 = new FutureTask<Integer>(new AccidentBatchLoader(40000,dataQueue2,fileQueue.remove()));
+                executor.execute(readerTask2);
+            }
+            Thread.sleep(1000*5);
+        }
+
+        while(!dataQueue1.isEmpty() || !dataQueue2.isEmpty()){
+            Thread.sleep(1000*5);
+        }
+        endDatLoading(executor);
+    }
+
+    public static void main(String[] args) throws Exception{
+        //handle2FilesWithRunnable();
+       //handle2FilesWithCallable();
+        handleMultipleFilesWithCallable();
+    }
+
+
+    private static class DataConsumer implements Runnable{
```

Comment:
private classes in Test class? :open_mouth: 

I'm pretty sure you have overcomplicated it :smile: